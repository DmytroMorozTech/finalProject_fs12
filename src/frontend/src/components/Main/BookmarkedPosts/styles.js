import {makeStyles} from '@material-ui/core/styles'

export default makeStyles((theme) => ({
  container: {
    marginTop: '75px'
  },

  savedElementsPanel: {
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white,
    padding: theme.spacing(2),
    paddingLeft: theme.spacing(4)
  },

  savedElementsHeading: {
    display: 'flex',
    alignItems: 'center',
    color: theme.palette.grey[700]
  },

  savedElementsSubheading: {
    paddingLeft: theme.spacing(1),
    paddingTop: theme.spacing(1)
  }
}))
