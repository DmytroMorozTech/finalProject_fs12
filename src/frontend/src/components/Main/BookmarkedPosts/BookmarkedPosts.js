import React from 'react'
import styles from './styles'
import { Container } from '@material-ui/core'
import Grid from '@material-ui/core/Grid'
import BookmarkIcon from '@material-ui/icons/Bookmark'
import Feed from '../Feed/Feed'

function BookmarkedPosts () {
  const classes = styles()

  return (
    <Container className={classes.container} maxWidth={'lg'}>
      <Grid container spacing={2} alignItems={'flex-start'} justifyContent={'center'}>
        <Grid item xs={12} sm={12} md={6} lg={6} xl={6}>
          <div className={classes.savedElementsPanel}>
            <h3 className={classes.savedElementsHeading}>
              <BookmarkIcon/>
              Saved Items
            </h3>
            <p className={classes.savedElementsSubheading}>Anything you save is private.</p>
          </div>

          <Feed type={'bookmarkedPosts'}/>
        </Grid>
      </Grid>
    </Container>
  )
}

export default BookmarkedPosts