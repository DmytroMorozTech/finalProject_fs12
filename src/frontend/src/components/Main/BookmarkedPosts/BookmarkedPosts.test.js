import { shallow } from 'enzyme'
import React from 'react'
import BookmarkedPosts from './BookmarkedPosts'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<BookmarkedPosts />)

describe('should render BookmarkedPosts component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
