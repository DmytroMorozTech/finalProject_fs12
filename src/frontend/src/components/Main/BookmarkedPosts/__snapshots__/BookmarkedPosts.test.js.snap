// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BookmarkedPosts component to match snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  maxWidth="lg"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justifyContent="center"
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={6}
      md={6}
      sm={12}
      xs={12}
    >
      <div>
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          <Memo(ForwardRef(BookmarkIcon)) />
          Saved Items
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="h5"
        >
          Anything you save is private.
        </WithStyles(ForwardRef(Typography))>
      </div>
      <Connect(Feed)
        type="bookmarkedPosts"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
